### ISSUE TITLE
breaking changes

### ISSUE THREAD

**Issue by ofirdagan** [tlvince/eslint-plugin-jasmine#124]
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
undefined exception

**If the current behavior is a bug, please provide the steps to reproduce.**
After this commit:
https://github.com/tlvince/eslint-plugin-jasmine/commit/99c458db02d5410de18da2f2b2e09a911c1704a4

My build gets this message:
`TypeError: Cannot read property 'start' of undefined
    at isFirstNodeInSuiteOrTest`

**What is the expected behavior?**

**Please mention your node.js, eslint-plugin-jasmine and operating system version.**
eslint-plugin-jasmine version: 2.6.0
node: 7.7.1

**Comment by ofirdagan** [comment]
@DianaSuvorova

**Comment by DianaSuvorova** [comment]
@ofirdagan, taking a look

**Comment by DianaSuvorova** [comment]
@ofirdagan, can you provide a snippet of code around which linter fails?

**Comment by ofirdagan** [comment]
@DianaSuvorova, checking

**Comment by ofirdagan** [comment]
@DianaSuvorova, ok found it.. we use an helper function in our tests to wrap async tests. The helper basically has try / catch and better error handling.
So a typical test will look like:
```
  it('should ...', specAsync(async () => {
    await uut.something('123');

    expect(uut.somethingElse).toEqual('123');
  }));
```

This break the lint..

**Comment by DianaSuvorova** [comment]
@ofirdagan , Thanks a lot for reporting this! 
I have reworked the  rule logic so it is much easier on assumptions about the structure of the code. And added your case to the tests. 

The issue you reported should be fixed with PR #126

**Comment by ofirdagan** [comment]
@DianaSuvorova Thank you for the quick fix!!
